### DISCUSSION TITLE
DMA Displaced Moving Average

### DISCUSSION THREAD

**Discussion by 200317314** [DaveSkender/Stock.Indicators#1102] [Ideas]
I was wondering if there was a way to add the DMA similar to TDAmeritrade 
https://tlc.thinkorswim.com/center/reference/Tech-Indicators/studies-library/C-D/DMA

To be able to use a negative displacement

**Comment by DaveSkender** [comment]
Hmm, interesting idea.  This sounds more like a utility than an actual indicator.  I imagine you might want to shift just about any indicator or moving average.  I'm thinking an interface like `quotes.GetSma(..).Shift(-3);` would work for anything, really.

By the way, this would be really simple to do yourself as well.

```csharp
// untested, just writing code from my iPad

var results = quotes.GetSma(..).ToList();

int shift = -3;
int length = results.Count();

for (int i = 0;  i < length;  i++) {

    // nullify out of range value
    if (i - shift < 0 || i - shift > length) {
        results[i].Sma = null;  // or remove record, if you prefer
    }   
 
    // transpose value
    else {
        results[i].Sma = results[i - shift].Sma ;
    }
}
```

**Reply by DaveSkender** [reply]
If interested, either writeup a new feature issue referencing this discussion, or create a PR with this utility if you want it sooner than later.  This will pair nicely with a set of math utilities in our backlog (see #979).
